src/Mailgun/Connection/RestClient.php

```diff
@@ -25,30 +25,59 @@ class RestClient
     private $apiKey;
 
     /**
-     * @var Guzzle
+     * @var HttpAdapter
      */
-    protected $mgClient;
+    protected $httpAdapter;
+
+    /**
+     * @var string
+     */
+    protected $apiEndpoint;
 
     /**
      * @param string $apiKey
-     * @param string $apiEndpoint
+     * @param string $apiHost
      * @param string $apiVersion
      * @param bool   $ssl
      */
-    public function __construct($apiKey, $apiEndpoint, $apiVersion, $ssl)
+    public function __construct($apiKey, $apiHost, $apiVersion, $ssl, HttpAdapter $adapter = null)
     {
         $this->apiKey = $apiKey;
-        $this->mgClient = new Guzzle([
-            'base_url'=>$this->generateEndpoint($apiEndpoint, $apiVersion, $ssl),
-            'defaults'=>[
-                'auth' => array(Api::API_USER, $this->apiKey),
-                'exceptions' => false,
-                'config' => ['curl' => [ CURLOPT_FORBID_REUSE => true ]],
```

Comment:
I understand that we need to set `CURLOPT_FORBID_RESUSE` for backwards compatibility, but why is that needed? I cant find any documentation about why we use it. This is from the [official documentation](http://curl.haxx.se/libcurl/c/CURLOPT_FORBID_REUSE.html)

> Pass a long. Set close to 1 to make libcurl explicitly close the connection when done with the transfer. Normally, libcurl keeps all connections alive when done with one transfer in case a succeeding one follows that can re-use them. This option should be used with caution and only if you understand what it does as it can seriously impact performance.

> Set to 0 to have libcurl keep the connection open for possible later re-use (default behavior).